benchmarks_gui/CMakeLists.txt

```diff
@@ -41,13 +51,17 @@ include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
 link_directories(${catkin_LIBRARY_DIRS})
 
-qt4_wrap_cpp(MOC_FILES include/main_window.h include/trajectory.h include/benchmark_processing_thread.h)
-qt4_wrap_ui(UIC_FILES
-  src/ui/main_window.ui
+set(MOC_FILES include/main_window.h include/trajectory.h include/benchmark_processing_thread.h)
```

Comment:
Automoc can handle a few cases automatically (https://cmake.org/cmake/help/v3.0/prop_tgt/AUTOMOC.html), however I think the ros case where headers are in a separate directory and there is no remark in source files about moc is not handled. Therefore I added all headers needing moc to MOC_FILES and added this as a "source" to the target.

So there are two possibilities (which might actually give identical results in this particular case)
a) MOC_HEADERS, a list of all headers in need of moc treatment
b) HEADERS, all headers
I'm fine with either one but personally use b) in private packages for better QtCreator support